packages/protocol/test-sol/governance/voting/LockedGold.t.sol

```diff
@@ -0,0 +1,2890 @@
+  }
+
+  function test_ShouldIncreaseTheAccountTOtalLockedGoldBalance() public {
+    lockedGold.lock.value(value)();
+    assertEq(lockedGold.getAccountTotalLockedGold(address(this)), value);
+  }
+
+  function test_ShouldIncreaseTheNonvotingLockedGoldBalance() public {
+    lockedGold.lock.value(value)();
+    assertEq(lockedGold.getNonvotingLockedGold(), value);
+  }
+
+  function test_ShouldIncreaseTheTotalLockedGoldBalance() public {
+    lockedGold.lock.value(value)();
+    assertEq(lockedGold.getTotalLockedGold(), value);
+  }
+
+  function test_ShouldEmitAGoldLockedEvent() public {
+    vm.expectEmit(true, true, true, true);
+    emit GoldLocked(address(this), value);
+    lockedGold.lock.value(value)();
+  }
+
+  function test_ShouldRevertWhenAccountDoesNotExist() public {
+    vm.expectRevert("Must first register address with Account.createAccount");
+    vm.prank(randomAddress);
+    lockedGold.lock();
+  }
+}
+
+contract Unlock is LockedGoldFoundryTest {
+  uint256 value = 1000;
+  uint256 availabilityTime = unlockingPeriod + block.timestamp;
+
+  uint256 votingGold = 1;
+  uint256 nonVotingGold = value - votingGold;
+
+  uint256 balanceRequirement = 10;
+
+  function setUp() public {
+    super.setUp();
+    lockedGold.lock.value(value)();
+  }
+
+  function test_ShouldAddAPendingWithdrawal() public {
+    lockedGold.unlock(value);
+    (uint256 val, uint256 timestamp) = lockedGold.getPendingWithdrawal(address(this), 0);
+    assertEq(val, value);
+    assertEq(timestamp, availabilityTime);
+    vm.expectRevert();
+    lockedGold.getPendingWithdrawal(address(this), 1);
+  }
+
+  function test_ShouldAddPendingWithdrawals() public {
+    lockedGold.unlock(value);
+    (uint256[] memory vals, uint256[] memory timestamps) = lockedGold.getPendingWithdrawals(
+      address(this)
+    );
+    assertEq(vals.length, 1);
+    assertEq(timestamps.length, 1);
+    assertEq(vals[0], value);
+    assertEq(timestamps[0], availabilityTime);
+  }
+
+  function test_ShouldDecreaseTheACcountsNonVotingLockedGoldBalance() public {
+    lockedGold.unlock(value);
+    assertEq(lockedGold.getAccountNonvotingLockedGold(address(this)), 0);
+  }
+
+  function test_ShouldDecreaseTheAccountsTotalLockedGoldBalance() public {
+    lockedGold.unlock(value);
+    assertEq(lockedGold.getAccountTotalLockedGold(address(this)), 0);
+  }
+
+  function test_ShouldDecreaseTheNonVotingLockedGoldBalance() public {
+    lockedGold.unlock(value);
+    assertEq(lockedGold.getNonvotingLockedGold(), 0);
+  }
+
+  function test_ShouldDecreaseTheTotalLockedGoldBalance() public {
+    lockedGold.unlock(value);
+    assertEq(lockedGold.getTotalLockedGold(), 0);
+  }
+
+  function test_ShouldEmitGoldUnlockedEvent() public {
+    vm.expectEmit(true, true, true, true);
+    emit GoldUnlocked(address(this), value, availabilityTime);
+    lockedGold.unlock(value);
+  }
+
+  function test_ShouldRevertWhenUnlockingGoldThatIsVotedWith() public {
+    governance.setVoting(address(this));
```

Comment:
Is there any reason we need this mock function and can not just simulate an actual vote?

Reply:
Most of our tests is designed as unit tests - meaning lots of interacting smart contracts are mocked. We could change our testing to integration tests/ e2e tests but it would require significant rewrite